src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -70,8 +73,19 @@ export class Main extends React.Component {
   }
 
   checkLoggedInStatus = () => {
+    const canCallSSO = this.props.useSSOe && !environment.isLocalhost();
+
+    if (canCallSSO) {
+      ssoKeepAliveSession();
+    }
+
     if (hasSession()) {
+      if (canCallSSO && !hasSessionSSO()) {
+        autoLogout();
+      }
```

Comment:
Idk if there's a better name for that function to make it more clear, since it's not implicit here, but the value of `hasSessionSSO` is derived from `ssoKeepAliveSession`. If it returns true, then `hasSessionSSO` is true and there's some additional logic about session expiration (it's different from vets-api and needs to be maintained for session continuity).

`ssoKeepAliveSession` is called independent of `hasSession` is for auto login. If the user doesn't have a `vets-api` session (ie, `hasSession` is `false`), but if they _do_ have an SSO session (`hasSessionSSO` is `true`) and can call SSO, we want to automatically redirect for a login.